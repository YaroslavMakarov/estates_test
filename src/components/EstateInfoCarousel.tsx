import React from 'react';
import { Row, Col,
         Carousel, CarouselItem } from 'react-bootstrap';
import { Img } from '../styled/estateStyles';   

type Props = {
    photos: Array<string>;
};

const EstateInfoCarousel: React.FC<Props> = ({ photos }) => (
    <Row>
        <Col sm={{span: 6, offset: 3}}>
            <Carousel>
                {photos.map(photo => (
                    <CarouselItem
                        key={photo}
                    >
                        <Img src={photo} />
                    </CarouselItem>
                ))}
            </Carousel>
        </Col>
    </Row>
);

export default EstateInfoCarousel;
