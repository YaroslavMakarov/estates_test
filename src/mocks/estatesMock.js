export const estatesMock = {
    estates: {
        estates: [
            {
                'id': 1,
                'title': 'Shevchenko estate',
                'price': 30000,
                'address': {
                    'street': 'Melnychna',
                    'number': 12
                },
                'preview_img': 'https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg',
                'seller': 'Chack Norris',
                'description': 'The best from Chuck',
                'detailed_photos': [
                    'https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg',
                    'https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg',
                    'https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg'
                ]
            },
            {
                "id": 2,
                "title": "Yanukovich estate",
                "price": 3000000,
                "address": {
                    "street": "Prisident",
                    "number": 4
                },
                "preview_img": "https://biz.liga.net/images/general/2018/11/15/thumbnail-tw-20181115111659-8474.jpg?v=1542279867",
                "seller": "Chack Norris",
                "description": "The best from Chuck",
                "detailed_photos": [
                    "https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg",
                    "https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg",
                    "https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg"
                    ]
             },
        ],
    },
};