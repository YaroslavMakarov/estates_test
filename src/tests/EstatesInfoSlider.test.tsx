import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';

import EstateInfoCarousel from '../components/EstateInfoCarousel';
import { estatesMock } from '../mocks/estatesMock';

describe('EstateInfoCarousel component', () => {
    let wrapper: RenderResult;
    const photos = estatesMock.estates.estates[0].detailed_photos

    beforeEach(() => {
        wrapper = render(
            <EstateInfoCarousel photos={photos}/>
        );
    });

    it('get EstateInfoCarousel snapshot', () => {
        expect(wrapper.baseElement).toMatchSnapshot();
    });

    it('check slides count', () => {
        expect(screen.getAllByAltText('slide')).toHaveLength(3);
    });

    it('check Img src attr', () => {
        const slides = screen.getAllByAltText('slide');

        slides.forEach((node, index) => {
            expect(node).toHaveAttribute('src', `${photos[index]}`)
        });
    });
});
