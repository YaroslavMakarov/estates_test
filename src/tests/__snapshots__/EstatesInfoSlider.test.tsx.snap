// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstateInfoCarousel component get EstateInfoCarousel snapshot 1`] = `
<body>
  <div>
    <div
      class="row"
    >
      <div
        class="col-sm-6 offset-sm-3"
      >
        <div
          class="carousel slide"
        >
          <ol
            class="carousel-indicators"
          >
            <li
              class="active"
            />
            <li />
            <li />
          </ol>
          <div
            class="carousel-inner"
          >
            <div
              class="active carousel-item"
            >
              <img
                alt="slide"
                class="sc-gsTCUz esHetz"
                src="https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg"
              />
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="slide"
                class="sc-gsTCUz esHetz"
                src="https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg"
              />
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="slide"
                class="sc-gsTCUz esHetz"
                src="https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg"
              />
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="sr-only"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="sr-only"
            >
              Next
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
